docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx

```diff
@@ -133,54 +130,174 @@ In the following table, we compare the supported methods for managing and passin
+
+Here's a look at how it works in Dagster Cloud:
+
+<center>
+  <Image
+    alt="Dagster Cloud encryption key hierarchy diagram"
+    src="/images/dagster-cloud/developing-testing/environment-variables/encryption-key-hierarchy.png"
+    width={836}
+    height={561}
+  />
+</center>
+
+In Dagster Cloud, each customer account is assigned a unique key, which then encrypts the data associated with that account. All customer keys are encrypted with a single, non-exportable AWS KMS master key.
+
+This approach isolates each account's data and reduces the risk of exposure by limiting the amount of data a single key can access.
+
+### Scope
+
+<Note>
+  This section is applicable only if using the Dagster Cloud UI to manage
+  environment variables.
+</Note>
+
+When creating a new environment variable in the Dagster Cloud UI, its scope will default to all deployments (full and [Branch Deployments](/dagster-cloud/developing-testing/branch-deployments)) and all [code locations](/dagster-cloud/developing-testing/code-locations)
+
+When assigning scope to an environment variable, keep the following in mind:
+
+- **For Branch Deployments**, variables will be read-only. Environment variables must be managed in the Branch Deployment's parent full deployment, which will usually be `prod`.
+
+- **You can create multiple instances of the same environment variable and assign different values and scopes.** Each instance of the variable can then be scoped to a deployment and code location (or locations). This approach can be useful for parameterizing behavior by environment without needing to modify your application code.
+
+  For example, let's say we want to use different database passwords in production and testing (Branch Deployments). In our code, we use the `SNOWFLAKE_PASSWORD` environment variable to pass in the database password. To use different passwords between production and Branch Deployments, we can create two instances of `SNOWFLAKE_PASSWORD`. One instance is scoped to the `prod` deployment and the other only to Branch Deployments:
+
+  <Image
+  alt="Example SNOWFLAKE_PASSWORD variables configured with different values based on deployment"
+  src="/images/dagster-cloud/developing-testing/environment-variables/same-variable-diff-scope.png"
+  width={1037}
+  height={188}
+  />
+
+  In this example, `SNOWFLAKE_PASSWORD=production_password` when in the `prod` deployment, while `SNOWFLAKE_PASSWORD=testing_password` in a Branch Deployment
+
+**Note**: If an environment variable isn't accessible at run time, [a `Variable not set` error will surface in Dagster Cloud](/guides/dagster/using-environment-variables-and-secrets).
```

Comment:
TBH it wasn't clear to me that this error was specific to that use case. Does this only come up then?

Reply:
Yeah, since environment variables can be accessed anywhere within your code we don't have a lot of control in general over how a failure resulting in one being missing will manifest

Reply:
Nope, nvm, got it from a comment further down. I'm just gonna take this out.